packages/front-end/components/Settings/EditDataSource/DataSourceInlineEditIdentityJoins/DataSourceInlineEditIdentityJoins.tsx

```diff
@@ -0,0 +1,224 @@
+        {identityJoins.map((identityJoin, idx) => {
+          const isOpen = openIndexes[idx] || false;
+          return (
+            <div className="bg-white border mb-3" key={`identity-join-${idx}`}>
+              <div className="d-flex justify-content-between">
+                {/* Title */}
+                <h4 className="py-3 px-3 my-0">
+                  {identityJoin.ids.join(" ↔ ")}
+                </h4>
+
+                {/* Actions*/}
+                <div className="d-flex align-items-center">
+                  <MoreMenu id="DataSourceInlineEditIdentifierTypes_identifier-joins">
+                    <button
+                      className="dropdown-item py-2"
+                      onClick={handleActionEditClicked(idx)}
+                    >
+                      <FaPencilAlt className="mr-2" /> Edit
+                    </button>
+
+                    <DeleteButton
+                      onClick={handleActionDeleteClicked(idx)}
+                      className="dropdown-item text-danger py-2"
+                      iconClassName="mr-2"
+                      style={{ borderRadius: 0 }}
+                      useIcon
+                      displayName={identityJoin.ids.join(" ↔ ")}
+                      deleteMessage={`Are you sure you want to delete identifier type ${identityJoin.ids.join(
+                        " ↔ "
+                      )}?`}
+                      title="Delete"
+                      text="Delete"
+                      outline={false}
+                    />
+                  </MoreMenu>
+
+                  <button
+                    className="btn ml-3"
+                    onClick={handleExpandCollapseForIndex(idx)}
+                  >
+                    <FaChevronRight
+                      style={{
+                        transform: `rotate(${isOpen ? "90deg" : "0deg"})`,
+                      }}
+                    />
+                  </button>
+                </div>
+              </div>
+              <div>
+                {isOpen && (
+                  <Code
+                    language="sql"
+                    theme="light"
+                    code={identityJoin.query}
+                    containerClassName="mb-0"
+                    expandable={true}
+                  />
+                )}
+              </div>
+            </div>
+          );
+        })}
+      </div>
+      {/* endregion Identity Joins list */}
+
+      {/* region Identity Joins empty state */}
+      {identityJoins.length === 0 ? (
+        <EmptyStateCard>
+          <div className="mb-3">
+            <h4>No identity joins.</h4>
+            {addIsDisabled ? (
+              <p>
+                You will be able to create identifier join tables when you have
+                identified at least 2 user identifiers.
+              </p>
+            ) : (
+              <p>
+                You can create identifier join tables with 2 or more identifiers
+              </p>
```

Comment:
Copy suggestions welcome. First part is supposed to be helpful text to tell the user how this section gets enabled.

The second one is just clarifying text explaining this section.

Reply:
I like the idea of an empty state when there's a clear CTA for the user. In this particular case, I don't think it makes a ton of sense when they don't have enough userIdTypes.

Creating an identity join on it's own is not a goal for users. Their goal is to be able to see experiment results. Identity join tables are just a means to that end and are only required if they have lots of userIdTypes and are mixing and matching them.

I would suggest hiding the entire IdentityJoins section until they have at least 2 userIdTypes. Then, always show it (empty state if there are no joins yet, otherwise the list).

Reply:
This suggestion is reverting back to the previous UX, which I had some concerns with:

1. it hides resources I've created, making them inaccessible
    1. when identifier types were deleted, it would hide the Identifier Joins section even though users had created some
    2. if a user is trying to "start over" and want to delete both identifier types and identifier joins, if they start from the top of the page and first delete the identifier types, they won't be able to delete the identifier joins
2. it hides functionality that can help a user evaluate or make the most use of the product
    1. if a user doesn't see a section, they may not know that feature exists.

Not sure if there's a way we can satisfy both sets of concerns. 🤔